app/cdap/services/datasource/DataSourceWebsockets.js

```diff
@@ -0,0 +1,391 @@
+/*
```

Comment:
Wonder if it's preferred to use ts?

Reply:
are these files copied from somewhere? maybe a previous pr before they removed websockets?

Reply:
This one is a copy, but `DataSourceHttp` is new and should be with us for a while. I'll see what it would take to add types.

Reply:
I've converted `DataSourceHttp` to TS